roles/libvirt_manager/tasks/start_manage_vms.yml

```diff
@@ -23,19 +23,22 @@
     index_var: vm_id
     label: "{{ vm_type }}-{{ vm_id }}"
 
+- name: "Get public network XML"
+  register: public_net_xml
+  community.libvirt.virt_net:
+    command: get_xml
+    name: "{{ public_net_nics[0].network }}"
+
 - name: "Grab IPs for nodes type {{ vm_type }}"  # noqa: risky-shell-pipe
   register: vm_ips
+  vars:
+    public_net_dict: "{{ public_net_xml.get_xml | ansible.utils.from_xml }}"
   ansible.builtin.shell:
     cmd: >-
-      virsh -c qemu:///system -q
-      domifaddr --source arp
-      cifmw-{{ nic.host }} | grep "{{ nic.mac }}"
-      {% if cifmw_openshift_api_ip_address | default(false) %}
-      | grep -v "{{ cifmw_openshift_api_ip_address }}/"
-      {% endif %}
-      {% if cifmw_openshift_ingress_ip_address | default(false) %}
-      | grep -v "{{ cifmw_openshift_ingress_ip_address }}/"
-      {% endif %}
+      ip -json neighbour show \
+        dev {{ nic.network }} \
+        to {{ public_net_dict.network.ip['@address'] }}/{{ public_net_dict.network.ip['@prefix'] }} \
+        | jq -r '.[] | select(has("lladdr")) | select(.["lladdr"] | contains ("{{ nic.mac }}")) | .dst'
```

Comment:
from what I see, it would be possible to `selectattr('lladdr', 'defined') | selectattr('lladdr', 'contains', nic.mac) | map(attribute='dst') | first` or something really, really close to this. Good for a follow up.